apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1470,6 +1520,27 @@ def create_network_postcommit(self, context):
             self.apic_manager.update_name_alias(
                 self.apic_manager.apic.fvAEPg, tenant_id, app_profile_name,
                 epg_name, nameAlias=network_name)
+            # set the tenant or application profile nameAlias if this is the
+            # first network being created under this tenant
+            if (context.current['tenant_id'] in
+                    self.tenants_with_name_alias_set):
+                return
+            tenant_name_alias = self.name_mapper.get_tenant_name(
+                context.current['tenant_id'], require_keystone_session=True)
+            if not tenant_name_alias:
+                return
+            if isinstance(tenant_id, apic_mapper.ApicName):
```

Comment:
I was planning to do that however if you look at the _get_network_aci_tenant() call, it could return tenant "common" if the network is shared. So it can't just depend on that flag alone.....That's why I'm doing this way.